Formula/cmusfm.rb

```diff
@@ -0,0 +1,76 @@
+class Cmusfm < Formula
+  desc "Last.fm standalone scrobbler for the cmus music player"
+  homepage "https://github.com/Arkq/cmusfm"
+  url "https://github.com/Arkq/cmusfm/archive/v0.3.3.tar.gz"
+  sha256 "9d9fa7df01c3dd7eecd72656e61494acc3b0111c07ddb18be0ad233110833b63"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "pkg-config" => :build
+
+  libfaketime = Formula["libfaketime"]
+  can_fake_time = libfaketime.requirements.all?(&:satisfied?)
+  depends_on "libfaketime" => :test if can_fake_time
+
+  def install
+    system "autoreconf", "--install"
+    mkdir "build" do
+      system "../configure", "--prefix=#{prefix}", "--disable-dependency-tracking", "--disable-silent-rules"
+      system "make", "install"
+    end
+  end
+
+  test do
+    cmus_home = testpath/".config/cmus"
+    cmusfm_conf = cmus_home/"cmusfm.conf"
+    cmusfm_sock = cmus_home/"cmusfm.socket"
+    cmusfm_cache = cmus_home/"cmusfm.cache"
+    faketime_conf = testpath/".faketimerc"
+
+    test_artist = "Test Artist"
+    test_title = "Test Title"
+    test_duration = can_fake_time ? 260 : 31
+    status_args = %W[
+      artist #{test_artist}
+      title #{test_title}
+      duration #{test_duration}
+    ]
+
+    mkpath cmus_home
+    touch cmusfm_conf
+
+    begin
+      server = fork do
+        if can_fake_time
+          faketime_conf.write "+0"
+          ENV["DYLD_INSERT_LIBRARIES"] = libfaketime.lib/"faketime"/"libfaketime.1.dylib"
+          ENV["DYLD_FORCE_FLAT_NAMESPACE"] = "1"
+          ENV["FAKETIME_NO_CACHE"] = "1"
+        end
+        exec bin/"cmusfm", "server"
+      end
+      loop do
+        sleep 0.5
+        assert_equal nil, Process.wait(server, Process::WNOHANG)
+        break if cmusfm_sock.exist?
+      end
+
+      system bin/"cmusfm", "status", "playing", *status_args
+      if can_fake_time
+        sleep 5
+        faketime_conf.atomic_write "+#{test_duration}"
+      else
+        sleep test_duration
+      end
+      system bin/"cmusfm", "status", "stopped", *status_args
+    ensure
+      Process.kill :TERM, server
+      Process.wait server
+    end
```

Comment:
Please forgive me for rehashing this, but: Is it not possible to create a test that tests some reasonable functionality without this level of complexity in the test?

Ideally, we want the test to run quickly so it doesn't hold up bottling and CI testing. If `libfaketime` is functional on Sierra, High Sierra and Mojave, then, at a minimum, I think all of the conditional logic should be removed, and `libfaketime` utilized. If it is possible to write a simpler test, that does not require a long wait time, without introducing the `:test` dependency on `libfaketime` that would be even better.

Reply:
Given the [cookbook](https://github.com/Homebrew/brew/blob/1.8.4/docs/Formula-Cookbook.md#add-a-test-to-the-formula) demands that tests actually exercise the functionality of the application, I'm not sure if we can make things simpler (beyond gutting out the no-libfaketime fallback, as you've suggested).

Reply:
@puxlit yes, I am aware of both, and I'm not suggesting we remove exercising actual functionality; merely inquiring if there is a simpler way to test. I'll go with your expertise here, if you say there is no simpler test, then I have no objections, so long as we can remove the libfaketime logic, and reduce the time spent idle/sleeping as much as possible.